src/GridLineEstimator.cpp

```diff
@@ -118,6 +119,19 @@ GridLineEstimator::GridLineEstimator(
                             line_extractor_settings_.hough_rho_resolution,
                             line_extractor_settings_.hough_theta_resolution,
                             0.0);
+
+    if (expected_image_format == "RGB") {
+        hsv_conversion_constant_ = CV_RGB2HSV;
+        image_encoding_ = sensor_msgs::image_encodings::RGB8;
+    }
+    else if (expected_image_format == "RGBA") {
+        // HSV does not have a CV_RGBA2HSV constant defined
+        hsv_conversion_constant_ = CV_RGB2HSV;
```

Comment:
Does this work?

Reply:
Also why do we need this variable if it's the same in all cases?

Reply:
We don't but I did it for consistency. Also yes it works.